src/Lime.Protocol/Client/ClientChannelExtensions.cs

```diff
@@ -30,6 +28,8 @@ public static class ClientChannelExtensions
 
             var receivedSession = await channel.StartNewSessionAsync(cancellationToken).ConfigureAwait(false);
 
+            using var _ = receivedSession.Metadata?.StartActivity("Client.Channel.EstablishSession");
```

Comment:
We shouldn't start the activity before calling `StartNewSessionAsync`, above? That call sends the very fisrt envelope to a Lime Server.

Reply:
No, because the channel listener is established inside the start new session and then all received envelopes would share the same root context.. The start new session async is instrumented by the transport itself when the first session package is sent.. Maybe there is a better way but every attempt I ended in a bad context sharing...

Reply:
Ignore everything I said, I mixed two different subjects and different parts of the code.. You're correct, I just made the changes